camayoc/tests/qpc/api/v1/sources/test_sources_common.py

```diff
@@ -170,6 +226,45 @@ def test_update_with_bad_host(src_type, scan_host, cleanup):
     assert_source_update_fails(original_data, src)
 
 
+@pytest.mark.parametrize('src_type', QPC_SOURCE_TYPES)
+@pytest.mark.parametrize('scan_host', CREATE_DATA)
+def test_update_with_bad_exclude_host(src_type, scan_host, cleanup):
+    """Attempt to update valid source with an invalid excluded host.
+
+    :id: 62cc08ad-1149-4c2b-a4ad-0e4c07d10ff6
+    :description: Create valid {network, vcenter} source, and then attempt to
+        update it with an invalid excluded hostname option.
+    :steps:
+        1) Create valid credential and source
+        2) Update the source with an invalid excluded host
+    :expectedresults: Error codes are returned and the source is not updated.
+    """
+    src = gen_valid_source(cleanup, src_type, scan_host)
+    original_data = copy.deepcopy(src.fields())
```

Comment:
Thought for future improvement I see that `copy.deepcopy` is used when ever `obj.fields()` are called. That said it seems that `fields` method could return a copy already.

Reply:
After playing with this, you're right. Fields returns a bare data structure so deep copy isn't needed.